samplesheets/views_api.py

```diff
@@ -421,3 +422,49 @@ def get(self, request, **kwargs):
             except Exception as ex:
                 return Response(str(ex), status=500)
         return Response(ret, status=200)
+
+
+class ProjectIrodsFileListAPIView(
+    SODARAPIBaseProjectMixin, SODARAPIBaseMixin, APIView
+):
+    """
+    Return a list of files in the project's sample data repository.
+
+    **URL:** ``/samplesheets/api/irods/files/{Project.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Parameters:**
+
+    - ``project``: Project SODAR UUID
+
+    **Returns:**
+
+    - ``irods_data``: List of iRODS data objects
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.view_sheet'
+
+    def get(self, request, *args, **kwargs):
+        if not settings.ENABLE_IRODS:
+            raise APIException('iRODS not enabled')
+        irods_backend = get_backend_api('omics_irods')
+        if not irods_backend:
+            raise APIException('iRODS backend not enabled')
+        project = self.get_project(request=request, kwargs=kwargs)
+        self.check_object_permissions(request, project)
+        path = request.query_params.get('path')
```

Comment:
There was no requirement in the spec for setting a custom path. This can be removed as, at least for now, we are interested in the full file listing under the sample repository of the project.

Also, the method here is not protected against iRODS paths containing `/../`. That could be used to get access to read a collection outside of the bounds of the project where the user might not have actual rights. They could not actually access the files, just get a file listing, but still it is a serious security issue. While thinking of this I realized that the same thing actually works for a couple of existing views too. I'll add a separate ticket for sanitizing path input.